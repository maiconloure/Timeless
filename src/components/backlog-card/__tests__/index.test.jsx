import React, { useState as useStateMock } from 'react';
import BacklogCard from '../index';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';

jest.mock('react', () => ({
  ...jest.requireActual('react'),
  useState: jest.fn(),
}));

jest.mock('../../../utils/importAll', () => ({ icons: {} }));

describe('Snapshot', () => {
  it('renderer true', () => {
    useStateMock.mockImplementation(() => [true, () => {}]);

    const tree = renderer.create(<BacklogCard />).toJSON();

    expect(tree).toMatchSnapshot();
  });
});

describe('Events', () => {
  it('simulate closing', () => {
    const setStateSpy = jest.fn();

    useStateMock.mockImplementation(() => [true, setStateSpy]);

    const wrapper = mount(<BacklogCard />);
    wrapper.find('button').at(0).simulate('click');
    expect(setStateSpy).toHaveBeenCalledWith(false);
    expect(setStateSpy).toHaveBeenCalledTimes(1);
  });

  // it('simulate opening', () => {
  //   const setStateSpy = jest.fn();

  //   useStateMock.mockImplementation(() => [false, setStateSpy]);

  //   const wrapper = mount(<BacklogCard />);
  //   wrapper.find('button').at(0).simulate('click');
  //   expect(setStateSpy).toHaveBeenCalledWith(true);
  //   expect(setStateSpy).toHaveBeenCalledTimes(1);
  // });
});
