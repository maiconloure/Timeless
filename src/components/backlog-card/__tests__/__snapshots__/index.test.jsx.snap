// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot renderer true 1`] = `
<div>
  <div
    className="sc-dIsAE bRTquH"
    style={
      Object {
        "background": "rgba(58, 166, 242, 0.5)",
        "border": "none",
        "marginLeft": "0",
        "marginTop": "0",
        "padding": "20px",
        "width": "fit-content",
      }
    }
  >
    <div>
      <div
        className="sc-fFSRdu dxmWXw"
      >
        <button
          className="sc-crzoUp iPTyCq"
          onClick={[Function]}
        >
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            strokeWidth="0"
            style={
              Object {
                "color": undefined,
              }
            }
            viewBox="0 0 1024 1024"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M563.8 512l262.5-312.9c4.4-5.2.7-13.1-6.1-13.1h-79.8c-4.7 0-9.2 2.1-12.3 5.7L511.6 449.8 295.1 191.7c-3-3.6-7.5-5.7-12.3-5.7H203c-6.8 0-10.5 7.9-6.1 13.1L459.4 512 196.9 824.9A7.95 7.95 0 0 0 203 838h79.8c4.7 0 9.2-2.1 12.3-5.7l216.5-258.1 216.5 258.1c3 3.6 7.5 5.7 12.3 5.7h79.8c6.8 0 10.5-7.9 6.1-13.1L563.8 512z"
            />
          </svg>
        </button>
      </div>
    </div>
    <div
      className="sc-bqGHjH efLMGY"
      style={
        Object {
          "marginBottom": "0",
        }
      }
    >
      <h1
        className="sc-ksluoS hNkUcY"
        style={
          Object {
            "borderBottom": "2px solid #014D82",
            "color": "#014D82",
            "fontSize": "1.8rem",
            "paddingBottom": "8px",
          }
        }
      >
        Product Backlog
      </h1>
    </div>
    <div
      className="sc-hBMVcZ hBmSHu"
    >
      <div
        className="sc-bkbjAj iOUaTI"
      >
        <img
          className="sc-fnVYJo eKCOSX"
          onClick={[Function]}
          src=""
          title=""
        />
      </div>
      <div
        className="sc-iemXMA eFkVwr"
      >
        <div
          className="sc-jNnnWF ggoOtv"
        >
          <div
            className="sc-dPaNSN bfNwUl"
          >
            <div
              className="sc-iBzFoy fXaTjK"
            >
              <div
                className="sc-efHXLn kPzEyo"
                contentEditable="true"
                suppressContentEditableWarning={true}
              >
                DESCRIÇÃO
              </div>
              <div
                className="sc-cTJmaU tIgNv"
                contentEditable="true"
                suppressContentEditableWarning={true}
              >
                ...
              </div>
            </div>
            <div
              className="sc-bBjRzc cGyVdI"
            >
              <div
                className="sc-cOigif hwShjN"
              >
                <h2
                  className="sc-khIgXV edrdSF"
                >
                  ETIQUETAS
                </h2>
                <div
                  className="sc-hTRkEk egvMYA"
                >
                  #Sprint1
                </div>
                <div
                  className="sc-jgPznn jexUXP"
                >
                  <img
                    alt="Ícone de Ampulheta"
                    className="sc-gSYCTC erUfDP"
                    onClick={[Function]}
                  />
                  <span>
                    Tempo Execução
                     
                    <strong
                      contentEditable="true"
                      style={
                        Object {
                          "color": "red",
                        }
                      }
                      suppressContentEditableWarning={true}
                    >
                      00:00
                    </strong>
                  </span>
                </div>
                <div
                  className="sc-laZMyp jagPVZ"
                >
                  <img
                    alt="Ícone de Cronômetro"
                    className="sc-gSYCTC erUfDP"
                    onClick={[Function]}
                  />
                  <span>
                    Data Entrega 
                    <strong
                      style={
                        Object {
                          "color": "#ffbe00",
                        }
                      }
                    >
                      00/00/00
                    </strong>
                  </span>
                </div>
              </div>
              <div
                className="sc-ArjOu eMLHYn"
              >
                <h2
                  className="sc-khIgXV edrdSF"
                >
                  AÇÕES
                </h2>
                <div
                  className="sc-iNiQeE cCqoxt"
                >
                  <img
                    alt="Ícone de Porta"
                    className="sc-jffIyK hMTDTH"
                    onClick={[Function]}
                  />
                  <span>
                    Mover
                  </span>
                </div>
                <div
                  className="sc-iNiQeE cCqoxt"
                >
                  <img
                    alt="Ícone de Copiar"
                    className="sc-jffIyK hMTDTH"
                    onClick={[Function]}
                  />
                  <span>
                    Copiar
                  </span>
                </div>
                <div
                  className="sc-iNiQeE cCqoxt"
                >
                  <img
                    alt="Ícone de Excluir"
                    className="sc-jffIyK hMTDTH"
                    onClick={[Function]}
                  />
                  <span>
                    Excluir
                  </span>
                </div>
                <button
                  className="sc-eJoaVz cFzXYX"
                >
                  feedback
                </button>
              </div>
            </div>
          </div>
          <div
            className="sc-oefIU hQlggo"
          >
            <div
              className="sc-hhIhEF iwqTSM"
            >
              <h2
                className="sc-eirseW kikfOT"
              >
                <img
                  alt="Ícone de Lista"
                  className="sc-jffIyK hMTDTH"
                />
                <span>
                  Checklist
                </span>
              </h2>
              <div
                className="sc-lbVuaH klsAhm"
              >
                Teste
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
